test/rekt/features/kafkasource/data_plane.go

```diff
@@ -300,6 +310,114 @@ func dataPlaneDelivery(name string, data delivery) *feature.Feature {
 	return f
 }
 
+func mtDataPlaneDelivery() *feature.Feature {
+	name := "mt-tenant-test-plain"
+	f := feature.NewFeatureNamed("Delivery/" + name)
+	delivery := delivery{
+		message: kafkaMessage{
+			Key:     "0",
+			Payload: `{"value":711}`,
+		},
+		matchers: AllOf(
+			HasData([]byte(`{"value":711}`)),
+			HasExtension("key", "0"),
+		),
+	}
+
+	delivery.bootstrapServer = auths["plain"].bootstrapServer
+	delivery.auth = auths["plain"]
+
+	f.Setup("set kafka message", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "message", delivery.message)
+	})
+	f.Setup("set bootstrap server", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "bootstrapServer", delivery.bootstrapServer)
+	})
+
+	topicName := feature.MakeRandomK8sName("topic") // A k8s name is also a valid topic name.
+
+	f.Setup("set kafka topic name", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "topicName", topicName)
+	})
+
+	f.Setup("install kafka topic", kafkatopic.Install(topicName))
+
+	sinkName := feature.MakeRandomK8sName("sink")
+	f.Setup("set sink name", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "sinkName", sinkName)
+	})
+
+	f.Setup("install sink", eventshub.Install(sinkName, eventshub.StartReceiver))
+
+	ksopts := []kafkasource.CfgFn{
+		kafkasource.WithBootstrapServers([]string{delivery.bootstrapServer}),
+		kafkasource.WithTopics([]string{topicName}),
+		kafkasource.WithSink(&duckv1.KReference{
+			Kind:       "Service",
+			Name:       sinkName,
+			APIVersion: "v1",
+		}, ""),
+	}
+
+	f.Setup("install a kafka source", kafkasource.Install(name, ksopts...))
+	f.Requirement("kafka topic must be ready", kafkatopic.IsReady(topicName))
+	f.Requirement("kafka source must be ready", kafkasource.IsReady(name))
+	f.Assert("different KafkaSource exists on the same pod", mtSinkReceiveEvent(name, delivery.matchers))
+
+	return f
+}
+
+func mtSinkReceiveEvent(name string, matchers EventMatcher) func(ctx context.Context, t feature.T) {
+	return func(ctx context.Context, t feature.T) {
+		var message kafkaMessage
+		state.GetOrFail(ctx, t, "message", &message)
+
+		topicName := state.GetStringOrFail(ctx, t, "topicName")
+		sinkName := state.GetStringOrFail(ctx, t, "sinkName")
+
+		if test_mt_source == "1" {
+			time.Sleep(10 * time.Second)
+		}
+
+		kcopts := []kafkacat.CfgFn{
+			kafkacat.WithBootstrapServer(kafkaBootstrapUrlPlain),
+			kafkacat.WithTopic(topicName),
+			kafkacat.WithKey(message.Key),
+			kafkacat.WithHeaders(message.Headers),
+			kafkacat.WithPayload(message.Payload),
+		}
+		env := environment.FromContext(ctx)
+		matchers = AllOf(
+			func(have event.Event) error {
+				kafkaSourceClientSet, err := kafkaclientset.NewForConfig(kc.Config)
```

Comment:
You should be able to use the injection API to get the client from the context. Look under pkg/client/injection/client. You should be able to do something like kafkaclient.Get(ctx)